Hydrogen 1.0.0 Failed to Update in Atom

It was all good before until I tried to execute the latest update. Judging from the version no., this is probably a big one. The error message spit out by Atom can be seen below:

```
> zeromq@3.2.1 install C:\Users\PINGSH~1\AppData\Local\Temp\apm-install-dir-1161017-15732-526vvq\node_modules\Hydrogen\node_modules\zeromq
> prebuild --install --preinstall "npm run build:libzmq"


> Hydrogen@1.0.0 install C:\Users\PINGSH~1\AppData\Local\Temp\apm-install-dir-1161017-15732-526vvq\node_modules\Hydrogen
> npm rebuild --build-from-source

C:\Users\PINGSH~1\AppData\Local\Temp\apm-install-dir-1161017-15732-526vvq
`-- (empty)


(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
(node) warning: possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit.
Trace
    at TLSSocket.addListener (events.js:239:17)
    at TLSSocket.Readable.on (_stream_readable.js:673:33)
    at Request.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\npm-registry-client\lib\request.js:153:7)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at ClientRequest.<anonymous> (C:\Users\Pingshun Huang\AppData\Local\atom\app-1.12.3\resources\app\apm\node_modules\request\request.js:823:10)
    at emitOne (events.js:82:20)
    at ClientRequest.emit (events.js:169:7)
    at tickOnSocket (_http_client.js:502:7)
    at onSocketNT (_http_client.js:514:5)
'C:\Users\Pingshun' is not recognized as an internal or external command,
operable program or batch file.
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\PINGSH~1\AppData\Local\Temp\apm-install-dir-1161017-15732-526vvq\package.json'
npm WARN apm-install-dir-1161017-15732-526vvq No description
npm WARN apm-install-dir-1161017-15732-526vvq No repository field.
npm WARN apm-install-dir-1161017-15732-526vvq No README data
npm WARN apm-install-dir-1161017-15732-526vvq No license field.
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Users\\Pingshun Huang\\AppData\\Local\\atom\\app-1.12.3\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\Pingshun Huang\\AppData\\Local\\atom\\app-1.12.3\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\Pingshun Huang\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\Pingshun Huang\\.atom\\.apmrc" "install" "C:\\Users\\PINGSH~1\\AppData\\Local\\Temp\\d-1161017-15732-1dyhcjc\\package.tgz" "--runtime=electron" "--target=1.3.6" "--arch=ia32" "--global-style" "--msvs_version=2015"
npm ERR! node v4.4.5
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE

npm ERR! Hydrogen@1.0.0 install: `npm rebuild --build-from-source`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the Hydrogen@1.0.0 install script 'npm rebuild --build-from-source'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the Hydrogen package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm rebuild --build-from-source
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs Hydrogen
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls Hydrogen
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\PINGSH~1\AppData\Local\Temp\apm-install-dir-1161017-15732-526vvq\npm-debug.log
npm ERR! code 1
```

Hi @lgeiger 
I just tried the `--verbose` method, it still didn't work. However, the error message seems a bit different this time. The space in my user folder name may still be an issue here
```
> zeromq@3.2.1 install C:\Users\PINGSH~1\AppData\Local\Temp\apm-install-dir-1161019-22628-ml0yq3\node_modules\Hydrogen\node_modules\zeromq
> prebuild --install --preinstall "npm run build:libzmq"


> Hydrogen@1.0.0 install C:\Users\PINGSH~1\AppData\Local\Temp\apm-install-dir-1161019-22628-ml0yq3\node_modules\Hydrogen
> npm rebuild --build-from-source

'C:\Users\Pingshun' is not recognized as an internal or external command,
operable program or batch file.
C:\Users\PINGSH~1\AppData\Local\Temp\apm-install-dir-1161019-22628-ml0yq3
`-- (empty)

npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\PINGSH~1\AppData\Local\Temp\apm-install-dir-1161019-22628-ml0yq3\package.json'
npm WARN apm-install-dir-1161019-22628-ml0yq3 No description
npm WARN apm-install-dir-1161019-22628-ml0yq3 No repository field.
npm WARN apm-install-dir-1161019-22628-ml0yq3 No README data
npm WARN apm-install-dir-1161019-22628-ml0yq3 No license field.
npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Users\\Pingshun Huang\\AppData\\Local\\atom\\app-1.12.4\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\Pingshun Huang\\AppData\\Local\\atom\\app-1.12.4\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\Pingshun Huang\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\Pingshun Huang\\.atom\\.apmrc" "install" "C:\\Users\\PINGSH~1\\AppData\\Local\\Temp\\d-1161019-22628-1udtomu\\package.tgz" "--runtime=electron" "--target=1.3.9" "--arch=ia32" "--global-style" "--msvs_version=2015"
npm ERR! node v4.4.5
npm ERR! npm  v3.10.5
npm ERR! code ELIFECYCLE

npm ERR! Hydrogen@1.0.0 install: `npm rebuild --build-from-source`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the Hydrogen@1.0.0 install script 'npm rebuild --build-from-source'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the Hydrogen package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm rebuild --build-from-source
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs Hydrogen
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls Hydrogen
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\PINGSH~1\AppData\Local\Temp\apm-install-dir-1161019-22628-ml0yq3\npm-debug.log
npm ERR! code 1
failed
```

--- a/package.json
+++ b/package.json
@@ -30,7 +30,7 @@
     ]
   },
   "scripts": {
-    "install": "npm rebuild --build-from-source",
+    "postinstall": "npm rebuild zeromq --build-from-source",
     "build:docs": "markdox lib/plugin-api/hydrogen-provider.coffee lib/plugin-api/hydrogen-kernel.coffee -o PLUGIN_API.md"
   },
   "repository": "https://github.com/nteract/hydrogen",

